src/error.rs

```diff
@@ -347,17 +347,124 @@ impl IntoResponse for Error {
+                ResponseError {
+                    name: "decode".to_string(),
+                    message: "failed to decode body as bytes, please ensure your JSON body is encoded properly".to_string(),
+                },
+            ], vec![]),
+            // Figure out
+            // TODO: Error::InvalidOptionsProvided(_) => {}
+            Error::FromUtf8Error(_) => crate::handlers::Response::new_failure(StatusCode::BAD_REQUEST, vec![
+                ResponseError {
+                    name: "decode".to_string(),
+                    message: "failed to decode body as Utf8, please ensure your JSON body is encoded properly".to_string(),
+                },
+            ], vec![]),
+            Error::MultipartError(e) => crate::handlers::Response::new_failure(StatusCode::BAD_REQUEST, vec![
+                ResponseError {
+                    name: "body".to_string(),
+                    message: "failed to decode the multipart body".to_string(),
+                },
+                ResponseError {
+                    name: "multipart".to_string(),
+                    message: format!("{:?}", e),
+                },
+            ], vec![]),
+            Error::InvalidMultipartBody => crate::handlers::Response::new_failure(StatusCode::BAD_REQUEST, vec![
+                ResponseError {
+                    name: "body".to_string(),
+                    message: "multipart body did not conform to specification".to_string(),
+                },
+            ], vec![]),
+            Error::InvalidApnsType(t) => crate::handlers::Response::new_failure(StatusCode::BAD_REQUEST, vec![
+                ResponseError {
+                    name: "decoding_error".to_string(),
+                    message: format!("failed to decode apns type, \"{}\" is invalid", t),
+                },
+            ], vec![
+                ErrorField {
+                    field: "type".to_string(),
+                    description: "apns push type decoding failed".to_string(),
+                    location: ErrorLocation::Unknown,
+                }
+            ]),
+            Error::InternalServerError => crate::handlers::Response::new_failure(StatusCode::INTERNAL_SERVER_ERROR, vec![
                 ResponseError {
                     name: "unknown_error".to_string(),
                     message: "This error should not have occurred. Please file an issue at: https://github.com/walletconnect/echo-server".to_string(),
                 },
+            ], vec![]),
+            Error::GeoIpReader(_) | Error::BatchCollector(_) => crate::handlers::Response::new_failure(StatusCode::INTERNAL_SERVER_ERROR, vec![
                 ResponseError {
-                    name: "dbg".to_string(),
-                    message: format!("{e:?}"),
-                }
-            ], vec![])
+                    name: "o11y".to_string(),
+                    message: "Internal error monitoring the request".to_string(),
+                },
```

Comment:
curious: when is the `GeoIpReader` used?

Reply:
https://github.com/WalletConnect/echo-server/blob/90a3a7e984b2a9e8f3d7ddbbf3c1de42c4919fbe/src/handlers/register_client.rs#L105-L136